src/dotnet/APIView/APIViewWeb/Controllers/AutoReviewController.cs

```diff
@@ -0,0 +1,214 @@
+
+            if (review != null)
+            {
+                if (apiRevisions.Any())
+                {
+                    // Delete pending apiRevisions if it is not in approved state before adding new revision
+                    // This is to keep only one pending revision since last approval or from initial review revision
+                    var automaticRevisions = new Queue<APIRevisionListItemModel>(apiRevisions.Where(r => r.APIRevisionType == APIRevisionType.Automatic));
+                    var latestAutomaticAPIRevision = automaticRevisions.Peek();
+                    var comments = await _commentsManager.GetCommentsAsync(review.Id);
+
+                    while (
+                        automaticRevisions.Any() &&
+                        !latestAutomaticAPIRevision.IsApproved &&
+                        !await _apiRevisionsManager.IsAPIRevisionTheSame(latestAutomaticAPIRevision, renderedCodeFile) &&
+                        !comments.Any(c => latestAutomaticAPIRevision.Id == c.RevisionId))
+                    {
+                        await _apiRevisionsManager.SoftDeleteAPIRevisionAsync(user: User, apiRevision: latestAutomaticAPIRevision);
+                        latestAutomaticAPIRevision = automaticRevisions.Dequeue();
+                    }
+
+                    // We should compare against only latest revision when calling this API from scheduled CI runs
+                    // But any manual pipeline run at release time should compare against all approved revisions to ensure hotfix release doesn't have API change
+                    // If review surface doesn't match with any approved revisions then we will create new revision if it doesn't match pending latest revision
+
+                    if (compareAllRevisions)
+                    {
+                        foreach (var approvedAPIRevision in apiRevisions.Where(r => r.IsApproved))
+                        {
+                            if (await _apiRevisionsManager.IsAPIRevisionTheSame(approvedAPIRevision, renderedCodeFile))
+                            {
+                                return approvedAPIRevision;
+                            }
+                        }
+                    }
+
+                    if (await _apiRevisionsManager.IsAPIRevisionTheSame(latestAutomaticAPIRevision, renderedCodeFile))
+                    {
+                        apiRevision = latestAutomaticAPIRevision;
+                        createNewRevision = false;
+                    }
+                }
+            }
+            else
+            {
+                review = await _reviewManager.CreateReviewAsync(packageName: codeFile.PackageName, language: codeFile.Language, isClosed: false);
+            }
+
+            // Check if user is authorized to modify automatic review
+            await ManagerHelpers.AssertAutomaticReviewModifier(user: User, review: review, authorizationService: _authorizationService);
```

Comment:
This should be done before removing pending revisions to avoid deleting pending reviews by sending a request with unmatching codefile by user.

Reply:
I have moved it to before deleting the `apiRevision` and have converted it to `AutoAPIRevisionModifierRequirement`